src/validateAtoms.ts

```diff
@@ -0,0 +1,70 @@
+import { atom, SetStateAction, WritableAtom } from 'jotai';
+import { loadable } from 'jotai/utils';
+
+import type { CommonState } from './atomWithValidate';
+
+export type Validator = <Values extends Record<string, unknown>>(
+  values: Values,
+) => Promise<Values>;
+
+export type ValidatorState = {
+  isValid: undefined | boolean;
+  error: null | Error | unknown;
+  isValidating: undefined | boolean;
+};
+
+type AtomWithValidation<Value> = WritableAtom<
+  CommonState<Value>,
+  SetStateAction<Value>
+>;
+
+type LabeledAtoms<Value> = {
+  [k: string]: AtomWithValidation<Value>;
+};
+
+export const validateAtoms = <Value>(
+  labeledAtoms: LabeledAtoms<Value>,
+  validator: Validator,
+) => {
+  const baseAtom = atom(async (get) => {
+    // extract value from each atom and assign to the given key as label
+    const values = Object.fromEntries(
+      Object.entries(labeledAtoms).map(([k, v]) => {
+        const atomValue = get(v);
+
+        return [k, atomValue.value];
+      }),
+    );
+
+    return validator(values);
```

Comment:
```suggestion
    await validator(values);
    return values;
```

Oh, okay, this alone doesn't work because of loadable.
We'd need one more internal atom.

Reply:
So, if I understand correctly 
the `validation` output when used as 
```js
// validation  would have 
// validation.isValid,validation.isValidating,validation.values, validation.error ?  
const [validation] = useAtom(validationAtom)
```

Reply:
yes